src/plugins/kibana_usage_collection/server/collectors/cloud/detector/cloud_services.ts

```diff
@@ -11,7 +12,5 @@ import { GCP } from './gcp';
 
 /**
  * An iteratable that can be used to loop across all known cloud services to detect them.
- *
- * @type {Array}
  */
-export const CLOUD_SERVICES = [AWS, GCP, AZURE];
+export const CLOUD_SERVICES: [typeof AWS, typeof GCP, typeof AZURE] = [AWS, GCP, AZURE];
```

Comment:
Super-NIT: `export const CLOUD_SERVICES: CloudService[] = [AWS, GCP, AZURE];`? 

Otherwise, if each one of them are not correctly extending `CloudService`, the `typeof` won't alert us about that (at least not in this line). What do you think?

Reply:
ended up removing the list of singletons entirely as it wasn't really doing anything